tuf/api/metadata.py

```diff
@@ -1258,17 +1267,173 @@ def to_dict(self) -> Dict[str, Any]:
         return snapshot_dict
 
 
+class SuccinctHashDelegations:
+    """A container with information about a succinct hash delegation.
+
+    Succinct hash delegation describes delegations for all possible target paths
+    to a variable number of bins, the names of which are generated based on the
+    two class attributes.
+
+    Args:
+        hash_prefix_len: Number of bits between 1 and 32 used to separate the
+            different bins.
+        bin_name_prefix: Prefix of all bin names.
+        unrecognized_fields: Dictionary of all attributes that are not managed
+            by TUF Metadata API.
+
+    Raises:
+            ValueError, TypeError, AttributeError: Invalid arguments.
+    """
+
+    def __init__(
+        self,
+        hash_prefix_len: int,
+        bin_name_prefix: str,
+        unrecognized_fields: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        if hash_prefix_len <= 0 or hash_prefix_len > 32:
+            raise ValueError("hash_prefix_len must be between 1 and 32")
+        if not isinstance(bin_name_prefix, str):
+            raise ValueError("bin_name_prefix must be a string")
+
+        self.hash_prefix_len = hash_prefix_len
+        self.bin_name_prefix = bin_name_prefix
+
+        # Calculate the zero padding value based on the total number of bins.
+        # If hash_prefix_len = 8 then number_of_bins = 256 and zero_padding = 3
+        # meaning the third bin will have a suffix of "003".
+        self.number_of_bins = 2**hash_prefix_len
+        self.zero_padding = int(math.log10(self.number_of_bins)) + 1
```

Comment:
Ha that's smart!  Should give the same result as `len(str(self.number_of_bins))`, right? I wonder if using `len` doesn't make it slightly easier to understand (and might also be more performant). But I'm fine either way.

Minor naming nit: Maybe something like `suffix_width` or so makes more sense? Because the amount of needed zero padding actually is `(int(math.log10(self.number_of_bins)) + 1) - len(bin_index)`.

Reply:
it's still decimal though and not hex...

I'm sorry if I sound like a broken record but I'm not sure if this has been addressed anywhere.  Maybe I'm missing something but I'd like to see this addressed in a comment at least: the spec seems to define the bin names containing hexadecimal yet the PR uses decimal?

Reply:
You're right, TAP 15 says "*NAME_PREFIX-COUNT where COUNT is a **hexadecimal** value between 0 and 2^BIT_LENGTH-1*". So my suggestion would be `len(f"{self.number_of_bins:x}"))`.

Reply:
Oh...totally missed that...

Reply:
I realized that besides the calculation of the zero-padding I will have to change the calculation of `suffix` in `get_bin_name` API call in order to save the suffix as HEX.

Reply:
I assume the reason for using hex here is that it makes the binary math a little easier for humans to verify: if your targetpath hash begins with "a1d8...", then in a 16 bit succinct delegation the correct bin name should be "prefix-a1d8" -- at least that's how I've understood it